util/stop/stopper.go

```diff
@@ -153,71 +154,70 @@ func (s *Stopper) AddCloser(c Closer) {
 // goroutines launched to do periodic work and the kv/db.go gateway which
 // accepts external client requests.
 //
-// Returns false to indicate that the system is currently draining and
+// Returns an error to indicate that the system is currently draining and
 // function f was not called.
-func (s *Stopper) RunTask(f func()) bool {
+func (s *Stopper) RunTask(f func()) error {
 	file, line, _ := caller.Lookup(1)
 	key := taskKey{file, line}
 	if !s.runPrelude(key) {
-		return false
+		return errors.New("node stopped")
 	}
 	// Call f.
 	defer s.runPostlude(key)
 	f()
-	return true
+	return nil
 }
 
-// RunAsyncTask runs function f in a goroutine. It returns false when the
+// RunAsyncTask runs function f in a goroutine. It returns an error when the
 // Stopper is draining and the function is not executed.
-func (s *Stopper) RunAsyncTask(f func()) bool {
+func (s *Stopper) RunAsyncTask(f func()) error {
 	file, line, _ := caller.Lookup(1)
 	key := taskKey{file, line}
 	if !s.runPrelude(key) {
-		return false
+		return errors.New("node stopped")
 	}
 	// Call f.
 	go func() {
 		defer s.runPostlude(key)
 		f()
 	}()
-	return true
+	return nil
 }
 
-// RunLimitedAsyncTask runs function f in a goroutine, using the given
-// channel as a semaphore to limit the number of tasks that are run
-// concurrently to the channel's capacity. Blocks until the semaphore
-// is available in order to push back on callers that may be trying to
-// create many tasks. Returns false if the Stopper is draining and the
-// function is not executed.
-func (s *Stopper) RunLimitedAsyncTask(sem chan struct{}, f func()) bool {
+// RunLimitedAsyncTask runs function f in a goroutine, using the given channel
+// as a semaphore to limit the number of tasks that are run concurrently to
+// the channel's capacity. Blocks until the semaphore is available in order to
+// push back on callers that may be trying to create many tasks. Returns an
+// error if the Stopper is draining and the function is not executed.
```

Comment:
the last part of that sentence is off. `s/and the function/, in which case the function/`